olive/engine/engine.py

```diff
@@ -420,7 +440,7 @@ def run_no_search(
             )
 
         # initialize the search strategy
-        self.search_strategy.initialize(self.pass_search_spaces, input_model_id, objective_dict)
+        self.search_strategy.initialize(self.pass_flows_search_spaces, input_model_id, objective_dict)
```

Comment:
So, for the no search case, it only runs the first flow? We only call `next_step` once, so it would return the first flow. 

Actually, it looks like there might be a bug in the `run_no_search` mode from the beginning. `next_step` with `pass-by-pass` execution order would only return the first pass. It only behaved as expected when the execution order is `joint`.

Reply:
The potential bug is not because of this PR so it doesn't block it. 

But I am interested in whether only one flow or all flows are run in no search mode.

Reply:
Good catch. It is expected to be all flows.
I will fix this part. Thanks.

Reply:
Good catch. It is expected to be all flows.
I will fix this part. Thanks.

Reply:
Updated.